lib/garbage-deleter.js

```diff
@@ -0,0 +1,893 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright 2020 Joyent, Inc.
+ */
+
+//
+// This contains the code for GarbageDeleter objects which are used by the
+// garbage-deleter program to process garbage collector instructions and delete
+// the appropriate files on disk.
+//
+
+var fs = require('fs');
+var path = require('path');
+var util = require('util');
+
+var assert = require('assert-plus');
+var readline = require('readline');
+var vasync = require('vasync');
+var VError = require('verror').VError;
+
+var common = require('./common');
+
+var DEFAULT_CONCURRENT_DELETES = 10;
+var DEFAULT_MANTA_ROOT = '/manta';
+var DEFAULT_MAX_LINE_LENGTH = 2048;
+var DEFAULT_MAX_LINES = 1000;
```

Comment:
The important thing is that there's a limit. There's a TODO note in bin/garbage-deleter.js that we should consider putting these parameters in SAPI so they can be shared between the two. I'll try to get that done soon and update this PR.

Reply:
I'm easy on sharing the value. I wouldn't think they particularly need to be tunable, so just clear comments of a link would be fine with me.  Is this "1000" roughly related to this from garbage-dir-consumer.js?

```
    self.readBatchSize = self.config.options.record_read_batch_size || 666;
```

Reply:
yes. The 666 there is a placeholder designed to be obvious when the value's unset for now. I think we probably can start out setting the tunable to 200 since that's similar to what we're using in some production environments currently. I guess what we could do is set an upper bound on the tunable (i.e. the maximum amount of memory we're willing to use in garbage-deleter) and let the tunable only apply to the collector. This way garbage-deleter protects itself, but also we don't have problems if we lower the value at garbage-collector and have existing files in the queue on the mako.